server/channels/api4/post_local.go

```diff
@@ -3,10 +3,57 @@
 
 package api4
 
-import "net/http"
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/v8/channels/app"
+	"github.com/mattermost/mattermost/server/v8/channels/audit"
+)
 
 func (api *API) InitPostLocal() {
 	api.BaseRoutes.Post.Handle("", api.APILocal(getPost)).Methods(http.MethodGet)
-
 	api.BaseRoutes.PostsForChannel.Handle("", api.APILocal(getPostsForChannel)).Methods(http.MethodGet)
+	api.BaseRoutes.Post.Handle("", api.APILocal(localDeletePost)).Methods(http.MethodDelete)
+}
+
+func localDeletePost(c *Context, w http.ResponseWriter, r *http.Request) {
```

Comment:
Can we combine this function and `deletePost` to keep the code DRY?

Reply:
They're a bit different due to the exclusion of the permissions in the local function, not much gained by combining the other parts of the function